src/collapse/bpftrace.rs

```diff
@@ -0,0 +1,212 @@
+                    // Otherwise we're in the middle of a stack
+                    else {
+                        // If we're at the beginning of a new line, add an empty string to our
+                        // "stack"
+                        if is_beginning_of_line {
+                            vec.push(String::with_capacity(256));
+                        }
+                        // Write the current character to last string in our "stack"
+                        vec.last_mut().unwrap().push(c);
+                    }
+                }
+            }
+            if is_beginning_of_line {
+                is_beginning_of_line = false;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Consumes all whitespace, if any, until the beginning of a series of digits. Then consumes
+/// the series of digits, writing their ascii representation into the provided buffer. Returns
+/// the number of bytes written.
+fn consume_unsigned_integer(
+    buf: &mut [u8],
+    chars: &mut Peekable<impl Iterator<Item = char>>,
+) -> Result<usize, io::Error> {
+    // We're expecting a number; so if we run out of characters, something went wrong.
+    let mut next = match chars.peek() {
+        Some(next) => next,
+        None => {
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                "Parsing Error. Expected a number. Found '\n'.",
+            ));
+        }
+    };
+
+    // Consume and ignore all whitespace. Again, we're expecting a number; so if we run out
+    // of characters, something went wrong.
+    while next.is_whitespace() {
+        chars.next().unwrap();
+        next = match chars.peek() {
+            Some(c) => c,
+            None => {
+                return Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "Parsing Error. Expected a number. Found '\n'.",
+                ));
+            }
+        }
+    }
+
+    // Consume all digits, writing their ascii representation into the provided buffer
+    let mut index = 0;
+    while next.is_numeric() {
+        let c = chars.next().unwrap();
+        buf[index] = c as u8;
+        index += 1;
+        next = match chars.peek() {
+            Some(c) => c,
+            None => &'a',
+        };
+    }
+
+    // Return the number of bytes written
+    Ok(index)
+}
+
+#[cfg(test)]
+mod tests {
+    use std::env;
+    use std::fs::File;
+    use std::io::{self, Write};
+    use std::path::Path;
+    use std::process::{exit, Command};
+
+    use super::*;
+
+    const NUMBER_OF_TEST_INPUTS: usize = 3;
+
+    #[test]
+    fn test_bpftrace() {
+        let root_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+        let data_dir = Path::new(&root_dir)
+            .join("tests")
+            .join("data")
+            .join("bpftrace");
+        let perl = Path::new(&root_dir)
+            .join("flamegraph")
+            .join("stackcollapse-bpftrace.pl");
+
+        for i in 1..NUMBER_OF_TEST_INPUTS + 1 {
```

Comment:
I think I would probably instead just [walk the files in the directory](https://doc.rust-lang.org/std/fs/fn.read_dir.html).